compiler/rustc_ast_passes/src/errors.rs

```diff
@@ -824,3 +824,51 @@ pub(crate) struct MissingAbi {
     #[suggestion(code = "extern \"<abi>\"", applicability = "has-placeholders")]
     pub span: Span,
 }
+
+#[derive(Diagnostic)]
+#[diag(ast_passes_abi_custom_safe_foreign_function)]
+pub(crate) struct AbiCustomSafeForeignFunction {
+    #[primary_span]
+    pub span: Span,
+
+    #[suggestion(
+        ast_passes_suggestion,
+        applicability = "maybe-incorrect",
```

Comment:
I think this should be machine-applicable? The only reason it might not be, I guess, is if we think that `extern "custom"` was incorrectly applied, but they had to write it first in order for any sort of mechanical tooling to apply the suggestion.